src/main/java/org/myrobotlab/service/NeoPixel.java

```diff
@@ -26,501 +26,829 @@
-      // neopixel.setPixel(pix);
-      neopixel.sendPixel(pix);
-      // neopixel.setAnimation(NEOPIXEL_ANIMATION_LARSON_SCANNER, 255, 0, 0, 1);
-      // arduino.enableBoardStatus(true);
-      // neopixel.setAnimation(NEOPIXEL_ANIMATION_LARSON_SCANNER, 0, 255, 0, 1);
-      boolean done = true;
-      if (done) {
-        return;
-      }
-      Servo servo = (Servo) Runtime.start("servo", "Servo");
-      servo.attach(arduino, 5);
-      servo.moveTo(180.0);
-      sleep(2000);
-      // neopixel.setAnimation(NEOPIXEL_ANIMATION_LARSON_SCANNER, 200, 0, 0, 1);
+      Set<String> ret = new HashSet<>();
+      List<String> c = Runtime.getServiceNamesFromInterface("NeoPixelController");
+      ret.addAll(c);
+      controllers = ret;
+      broadcastState();
     } catch (Exception e) {
-      Logging.logError(e);
+      error(e);
     }
+    return controllers;
+  }
 
+  @Override
+  public void setAnimation(int animation, int red, int green, int blue, int wait_ms) {
+    if (wait_ms < 1) {
+      wait_ms = 1;
+    }
+    if (controller == null) {
+      error("%s could not set animation no attached controller", getName());
+      return;
+    }
+    log.info("setAnimation {} {} {} {} {}", animation, red, green, blue, wait_ms);
+    NeoPixelController nc2 = (NeoPixelController) Runtime.getService(controller);
+    nc2.neoPixelSetAnimation(getName(), animation, red, green, blue, 0, wait_ms);
   }
 
   @Override
-  public void setAnimation(int animation, int red, int green, int blue, int speed) {
-    // protect against 0 and negative speed
-    if (speed < 1)
-      speed = 1;
-    controller.neoPixelSetAnimation(this, animation, red, green, blue, speed);
-    this.animation = animationIntToString(animation);
-    broadcastState();
+  public void setAnimation(String animation, int red, int green, int blue, int wait_ms) {
+    // TODO convert to int
+
   }
 
-  String animationIntToString(int animation) {
-    switch (animation) {
-      case NEOPIXEL_ANIMATION_NO_ANIMATION:
-        return "No animation";
-      case NEOPIXEL_ANIMATION_STOP:
-        return "Stop";
-      case NEOPIXEL_ANIMATION_COLOR_WIPE:
-        return "Color Wipe";
-      case NEOPIXEL_ANIMATION_LARSON_SCANNER:
-        return "Larson Scanner";
-      case NEOPIXEL_ANIMATION_THEATER_CHASE:
-        return "Theater Chase";
-      case NEOPIXEL_ANIMATION_THEATER_CHASE_RAINBOW:
-        return "Theater Chase Rainbow";
-      case NEOPIXEL_ANIMATION_RAINBOW:
-        return "Rainbow";
-      case NEOPIXEL_ANIMATION_RAINBOW_CYCLE:
-        return "Rainbow Cycle";
-      case NEOPIXEL_ANIMATION_FLASH_RANDOM:
-        return "Flash Random";
-      case NEOPIXEL_ANIMATION_IRONMAN:
-        return "Ironman";
-      default:
-        log.error("Unknow Animation type {}", animation);
-        return "No Animation";
+  @Override
+  public void setAnimationSetting(String animation) {
+    // TODO Auto-generated method stub
+
+  }
+
+  public void setBlue(int blue) {

```

Comment:
Why doe we have something to set blue?  this method should probably be removed.

Reply:
These members 
red green blue ... represent a single color for which actions and control messages can be used with.
This is how it worked before - where an animation required a r,g,b value although in some cases it wasn't appropriate.
I separated part of this concept so that a "currently selected color" could be used ... and it is used when you select a color from the UI